File: count-primes/count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        
        if (n < 2) { return 0; }
        
        vector<int> isPrime(n, 1);
        
        isPrime[0] = 0;
        isPrime[1] = 0;
        
        for (int i = 2; i < sqrt(n); ++i) {
            
            if (isPrime[i] == 1) {
                
                for (long j = i*i; j < n; j += i) {
                    
                    isPrime[j] = 0;
                }
            }
        }
        
        return count (begin(isPrime), end(isPrime), 1);
    }
};

File: count-primes/count-primes.java
class Solution {
    public int countPrimes(int n) {
        
        if (n < 2) { return 0; }
        
        int[] notPrime = new int[n];
        
        notPrime[0] = 1;
        notPrime[1] = 1;
        
        for (int i = 2; i < Math.sqrt(n); ++i) {
            
            if (notPrime[i] == 0) {
                
                for (int j = i*i; j < n; j += i) {
                    
                    notPrime[j] = 1;
                }
            }
        }
        
        int count = 0;
        
        for (int i : notPrime) {
            
            if (i == 0) { ++count; }
        }
        
        return count;
    }
}